zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -14,193 +14,240 @@
         pk = [y2s0, y2s1, y2s2, y2s3, y2s4]
-        # channel ID, close flag, rev lock, Cust Bal and Merch Bal
+        # the message is composed of the channel ID, close flag, rev lock, cust closing balance and 
+        # merchant closing balance.
         msg = [cid, close_b, rev_lock_b.value, cust_bal_b.value, merch_bal_b.value]
         prod1 = sp.local('prod1', x2)
         for i in range(0, len(msg)):
             prod1.value += sp.mul(pk[i], msg[i])
+        # Compute the pairing check.
         sp.verify(sp.pairing_check([sp.pair(s1, prod1.value), sp.pair(s2, -g2)]), message="pairing check failed")
-        # Update on-chain state and transfer merchant's balance   
+        ## End of PS signature verification
+
+        # Update the closing balances in the contract storage.  
         self.data.custBal = custBal
         self.data.revLock = revLock
+        # Set delayExpiry to the current time plus the specified delay period, selfDelay. When 
+        # delayExpiry has passed, the customer will be able to claim their balance by calling 
+        # custClaim. 
         self.data.delayExpiry = sp.now.add_seconds(self.data.selfDelay)
-        # Pay merchant immediately (unless amount is 0)
-        # Note that all addresses must be implicit accounts (tz1), not smart contracts
+        # Pay merchant immediately (unless amount is 0).
         sp.if merchBal != sp.tez(0):
             sp.send(self.data.merchAddr, merchBal)
-        self.data.merchBal = sp.tez(0)
+        # Set the channel status to CUST_CLOSE.
         self.data.status = CUST_CLOSE
  
-    # merchDispute can be called if the merchant has the secret corresponding
-    # to the latest custClose state. If the secret is valid, the merchant will
-    # receive the customer's balance too.
+    # merchDispute can be called by the merchant if they have the revocation secret corresponding 
+    # to the revocation lock in the customer's custClose entrypoint call. If the secret is valid, 
+    # the merchant will receive the customer's balance.
     @sp.entry_point
     def merchDispute(self, secret):
+        # Only allow the merchant to call the entrypoint.
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is CUST_CLOSE.
         sp.verify(self.data.status == CUST_CLOSE)
         
-        # convert rev_lock in storage from LE to BE
+        # Convert rev_lock in storage from LE to BE
         revlock_be = sp.local('revlock_be', sp.list([]))
         sp.for i in sp.range(0, 32):
             revlock_be.value.push(sp.slice(self.data.revLock, i, 1).open_some())
-            
+        # Verify the revocation secret hashes to the revocation lock
         sp.verify(sp.concat(revlock_be.value) == sp.sha3(secret))
+        # Send the customer's (revoked) balance to the merchant
         sp.send(self.data.merchAddr, self.data.custBal)
-        self.data.custBal = sp.tez(0)
         self.data.status = CLOSED
```

Comment:
add documentation for L225 for completeness.

Reply:
added